src/cosmic_ray/commands/execute.py

```diff
@@ -30,30 +30,29 @@ def _report_progress(stream):
 
 
 @reports_progress(_report_progress)
-def execute(db_name):
+def execute(work_db):
     """Execute any pending work in the database stored in `db_name`,
     recording the results.
 
     This looks for any work in `db_name` which has no results, schedules it to
     be executed, and records any results that arrive.
     """
     try:
-        with use_db(db_name, mode=WorkDB.Mode.open) as work_db:
+        _update_progress(work_db)
+        config = work_db.get_config()
+        engine = get_execution_engine(config.execution_engine_name)
+
+        def on_task_complete(job_id, work_result):
+            work_db.set_result(job_id, work_result)
             _update_progress(work_db)
-            config = work_db.get_config()
-            engine = get_execution_engine(config.execution_engine_name)
-
-            def on_task_complete(job_id, work_result):
-                work_db.set_result(job_id, work_result)
-                _update_progress(work_db)
-                log.info("Job %s complete", job_id)
-
-            log.info("Beginning execution")
-            engine(
-                work_db.pending_work_items,
-                config,
-                on_task_complete=on_task_complete)
-            log.info("Execution finished")
+            log.info("Job %s complete", job_id)
+
+        log.info("Beginning execution")
+        engine(
+            work_db.pending_work_items,
+            config,
+            on_task_complete=on_task_complete)
+        log.info("Execution finished")
 
     except FileNotFoundError as exc:
```

Comment:
Where is this `FileNotFoundError` going to come from? It used to come from `use_db`, but that's not an issue now.

Reply:
Removed. I don't re-raise anywhere `FileNotFoundError`, but I had changed message of the 1st.